torch/_functorch/aot_autograd.py

```diff
@@ -1324,34 +1324,45 @@ def aot_dispatch_base(flat_fn, flat_args: List[Tensor], aot_config: AOTConfig, *
             flat_args = (seed, offset, *flat_args)
         compiled_fw = compiler(fw_module, flat_args)
 
-    # Get the RNG functionalization related metadata to be used at runtime.
-    rng_metadata = RNGMeta(False, PhiloxTotalOffsets(0, 0))
-    if config.functionalize_rng_ops:
-        rng_metadata = RNGMeta(config.functionalize_rng_ops, PhiloxStateTracker.get_accumulated_offsets())
+    # This boxed_call handling happens inside create_runtime_wrapper as well.
+    # However, create_runtime_wrapper does not expect the rng offsets in the
+    # output. So, we have to create another wrapper and take out the offset. As
+    # a result, we have to account for not boxed_call compilers as well.
+    if not hasattr(compiled_fw, "_boxed_call"):
+        compiled_fw = make_boxed_func(compiled_fw)
+
+    # Create a wrapper to set up the rng functionalize bits
+    @wraps(compiled_fw)
+    def rng_functionalization_wrapper(args):
+        # args is a list because compiled_fw is boxed_call
+        if fw_metadata.is_rng_op_functionalized:
+            # Add the seed and offset to args
+            seed, offset = CUDARngStateHelper.get_torch_state_as_tuple()
+            new_args = [seed, offset, *args]
```

Comment:
This is bug, you need to clear args here otherwise you will keep the inputs live too long

Reply:
IIUC, we call `.clear` inside inductor. But not calling `.clear` here would create an extra reference, rendering `.clear` call inside Inductor ineffective because memory will still be held here. Is that correct?

I changed it other places as well.

Really nice catch!

Reply:
new_args would get cleared, but not args (because `*args` causes a copy)